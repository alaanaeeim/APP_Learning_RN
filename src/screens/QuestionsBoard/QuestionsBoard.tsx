import * as React from 'react';
import {
  View,
  ScrollView,
  StyleSheet,
  Text,
  TextInput,
  Button,
  Image,
} from 'react-native';

const QuestionsBoard = () => {
  const [username, setUsername] = React.useState('Alaa Naeeim');
  return (
    <ScrollView>
      <View>
        <Text accessibilityLabel="usernameText">{username}</Text>

        <TextInput
          accessibilityLabel="username"
          placeholder="Enter USername"
          value={username}
          onChangeText={val => setUsername(val)}
        />

        <Image
          style={styles.image}
          source={require('../../assets/images/iphone.jpg')}
        />
        <Button
          title="Change Username"
          accessibilityLabel="change username"
          onPress={() => setUsername('Bassem Naeeim')}
        />
      </View>
    </ScrollView>
  );
};

export default QuestionsBoard;

const styles = StyleSheet.create({
  image: {
    width: 100,
    height: 100,
    resizeMode: 'contain',
  },
});
