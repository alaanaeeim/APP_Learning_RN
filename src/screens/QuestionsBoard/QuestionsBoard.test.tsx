import React from 'react';
import {fireEvent, render} from '@testing-library/react-native';
import QuestionsBoard from './QuestionsBoard';

jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');

test('Test QuestionsBoard Screen', () => {
  const {toJSON, getByLabelText, getAllByLabelText} = render(
    <QuestionsBoard />,
  );

  expect(toJSON()).toMatchSnapshot();
  const helloText = getByLabelText('usernameText');
  const allInputs = getAllByLabelText('username');
  expect(helloText.props.children).toBe('Alaa Naeeim');

  fireEvent.changeText(allInputs[0], 'Mina Naeeim');
  expect(helloText.props.children).toBe('Mina Naeeim');

  const changeUsernameBtn = getByLabelText('change username');
  fireEvent.press(changeUsernameBtn);
  expect(helloText.props.children).toBe('Bassem Naeeim');

  console.log(helloText.children);
});
