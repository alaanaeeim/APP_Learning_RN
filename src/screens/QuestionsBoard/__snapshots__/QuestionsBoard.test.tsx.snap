// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test QuestionsBoard Screen 1`] = `
<RCTScrollView>
  <View>
    <View>
      <Text
        accessibilityLabel="usernameText"
      >
        Alaa Naeeim
      </Text>
      <TextInput
        accessibilityLabel="username"
        onChangeText={[Function]}
        placeholder="Enter USername"
        value="Alaa Naeeim"
      />
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/images/iphone.jpg",
          }
        }
        style={
          Object {
            "height": 100,
            "resizeMode": "contain",
            "width": 100,
          }
        }
      />
      <View
        accessibilityLabel="change username"
        accessibilityRole="button"
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Change Username
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
